kf/bpf.go

```diff
@@ -65,13 +65,31 @@ type BPF struct {
 	Ctx               context.Context           `json:"-"`
 	Done              chan bool                 `json:"-"`
 	ProgMapCollection *ebpf.Collection          `json:"_"` // eBPF Collection reference
-	ProgMapID         ebpf.MapID
+	ProgMapID         ebpf.MapID                // Prog map id
+	PrevProgMapID     ebpf.MapID                // Prev prog map id
 	hostConfig        *config.Config
 	TCFilter          *tc.Filter `json:"-"` // handle to tc filter
 	XDPLink           link.Link  `json:"-"` // handle xdp link object
 }
 
-func NewBpfProgram(ctx context.Context, program models.BPFProgram, conf *config.Config) *BPF {
+func NewBpfProgram(ctx context.Context, program models.BPFProgram, conf *config.Config, ifaceName string) *BPF {
+
+	var progMapFilePath string
+	if len(program.MapName) > 0 {
+		if program.ProgType == models.XDPType {
+			progMapFilePath = filepath.Join(conf.BpfMapDefaultPath, ifaceName, program.MapName)
+		} else if program.ProgType == models.TCType {
+			ss := strings.Split(program.MapName, "/")
```

Comment:
for TC we treat this as a path vs XDP we treat it as a name.  Is this expected?

Reply:
In case TC map names are provided `tc/globals/<mapname>` (i.e. in R1). Also we can skip the path and provide mapname only, it handles both ways.

Reply:
can we make the behavior the same across them then?  XDP and TC where you can provide map name or full path of map name.